### ISSUE TITLE
Posting messages to WebView fails

### ISSUE THREAD

**Issue by llakie** [microsoft/vscode#127460]
Issue Type: <b>Bug</b>

- Create a WebviewPanel and enable scripts
- Implement the `onDidReceiveMessage` of the Webview
- Implement `addEventListener('message', msg => { console.log(msg); })` in your HTML
- Post a message from the class where you instantiated the WebviewPanel

Expected behavior: message callback on Webview is called. Here `msg => console.log(msg);`
Actual behavior: `onDidReceiveMessage` is called

If you add the `onMessage` method using the console provided by the `Webview developer tools` after the Webview has been loaded, it receives the message with a lot of overhead around the actual message payload being sent. I just copied the interesting part, where the payload is nested at `data.data.message` instead of the expected `data` field.
```json
{
  "data": {
    "channel": "onmessage",
    "data": {
      "message": {
        "action": "publish",
        "source": "controller",
        "success": true
      }
    },
    "target": "66082911-0010-43a7-8536-9ad9c1432797"
  },
  "defaultPrevented": false  
}
```

I unfortunately cannot provide the specific version of VSCode, when it stopped working.

VS Code version: Code 1.57.1 (507ce72a4466fbb27b715c3722558bb15afa9f48, 2021-06-17T13:28:07.755Z)
OS version: Windows_NT x64 10.0.18363
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-6300U CPU @ 2.40GHz (4 x 2496)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.88GB (16.46GB free)|
|Process Argv|--crash-reporter-id 1a1bb333-20a6-4af5-ba14-2fe0663f914c|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (31)</summary>

Extension|Author (truncated)|Version
---|---|---
jslint|ajh|1.2.1
iotea|bos|0.9.5
vscode-markdownlint|Dav|0.42.1
vscode-eslint|dba|2.1.23
gitlens|eam|11.5.1
vscode-npm-script|eg2|0.3.22
markdown-preview-bitbucket|hbr|1.0.0
beautify|Hoo|1.5.0
workspacesort|ici|1.6.0
plantuml|jeb|2.15.1
jwt-decoder|jfl|1.2.0
vscode-azurefunctions|ms-|1.4.0
vscode-azureresourcegroups|ms-|0.4.0
vscode-docker|ms-|1.14.0
python|ms-|2021.6.944021595
vscode-pylance|ms-|2021.6.3
jupyter|ms-|2021.6.999662501
remote-containers|ms-|0.183.0
remote-wsl|ms-|0.56.5
azure-account|ms-|0.9.8
azurecli|ms-|0.5.0
vscode-typescript-tslint-plugin|ms-|1.3.3
vsliveshare|ms-|1.0.4498
debugger-for-chrome|msj|4.12.12
java|red|0.79.2
vscodeintellicode|Vis|1.2.14
vscode-java-debug|vsc|0.34.0
vscode-java-dependency|vsc|0.18.5
vscode-java-pack|vsc|0.17.0
vscode-java-test|vsc|0.30.1
vscode-maven|vsc|0.31.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30300192
pythontb:30283811
vspre833:30321513
vshan820:30294714
vscus158:30321503
pythonvsuse255cf:30323309
vscorehov:30309549
vscod805cf:30301675
binariesv615:30325510

```

</details>

**Comment by mjbvz** [comment]
Please share a minimal example extension that demonstrates the issue

**Comment by llakie** [comment]
I solved it. I did not realize, that I hat a function in one of my JS files called `postMessage` (which in this case overrides `window.postMessage()`). Removing this function solves the issue completly. But one question remains: Why did it work with an older version of VSCode? Did you exchange the extension runtime at some point in the recent past, or did you change to use the postMessage API for communication with the Extension Code recently? Thanks for answering my question.
